### PR TITLE
Replace Accompanist Systemuicontroller with edge-to-edge

### PR DESCRIPTION
**Pull request by wmontwe**
This replaces the deprecated [Accompanist - System UI Controller](https://google.github.io/accompanist/systemuicontroller/) with the new [edge-to-edge](https://developer.android.com/develop/ui/views/layout/edge-to-edge) feature of the AndroidX Activity library.

Depends on #7660

### CODE REVIEW

File: core/ui/compose/theme/src/main/java/app/k9mail/core/ui/compose/theme/color/SystemBarsColor.kt [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
 ) {
-    val systemUiController = rememberSystemUiController()
-
-    DisposableEffect(Unit) {
-        systemUiController.setSystemBarsColor(color = color)
-
-        onDispose { }
+    val view = LocalView.current
+    if (!view.isInEditMode) {
+        SideEffect {
```
Comment (cketti) [inline core/ui/compose/theme/src/main/java/app/k9mail/core/ui/compose/theme/color/SystemBarsColor.kt:18]: Why this change from `DisposableEffect`? With `SideEffect` the lambda is executed on every recomposition. 

(In practice this doesn't seem to make a difference right now.)

**Review by cketti — APPROVED** [review]

File: core/ui/compose/theme/src/main/java/app/k9mail/core/ui/compose/theme/color/SystemBarsColor.kt (hunk 1 above)
Comment (wmontwe) [reply core/ui/compose/theme/src/main/java/app/k9mail/core/ui/compose/theme/color/SystemBarsColor.kt:18]: My understanding is the following:

`SideEffect` is executed when its parent Composable is recomposed -> as it's part of the theme it will react to theme changes or activty related recompositions. So it's not critical.

But if you would like to reduce recompositions to a minimum `DisposableEffect` and a key of `darkTheme`and `color` would be indeed better.

I think in the theme, the impact is barely noticable. If you would like to control the status bar from a Composable deeper in the hierarchy it will recompose too often. I'll update the approach.

Comment (wmontwe) [reply core/ui/compose/theme/src/main/java/app/k9mail/core/ui/compose/theme/color/SystemBarsColor.kt:18]: Ah I remember why I picked SideEffect, it's lambda is not of composable type and requires some tweaks:

```
@Composable
fun SetSystemBarsColor(
    darkTheme: Boolean,
    color: Color = Color.Transparent,
) {
    val view = LocalView.current

    DisposableEffect(view, darkTheme, color) {
        if (!view.isInEditMode) {
            val window = (view.context as Activity).window
            window.statusBarColor = color.toArgb()
            WindowCompat.getInsetsController(window, view).isAppearanceLightStatusBars = !darkTheme
        }
        onDispose { }
    }
}
```

Comment (wmontwe) [reply core/ui/compose/theme/src/main/java/app/k9mail/core/ui/compose/theme/color/SystemBarsColor.kt:18]: I will stick with the original version
